overlays/ghc-packages.nix

```diff
@@ -83,12 +84,12 @@ let
 
   # The nix produced by `cabalProject` differs slightly depending on
   # what the platforms are.  There are currently 3 possible outputs.
-  ghc-extra-projects-type =
+  ghc-extra-projects-type = ghc:
     if final.stdenv.hostPlatform.isWindows
       then "windows"
       else if final.stdenv.hostPlatform.isGhcjs
         then "ghcjs"
-        else if final.haskell-nix.haskellLib.isCrossHost
+        else if final.haskell-nix.haskellLib.isCrossHost || (final.stdenv.hostPlatform.isMusl && builtins.compareVersions ghc.version "9.4.1" >= 0)
```

Comment:
Huh? I don't understand why this should lead to cross?
`(final.stdenv.hostPlatform.isMusl && builtins.compareVersions ghc.version "9.4.1" >= 0)`

Reply:
I might rename those from `default` and `cross` to `terminfo` and `noterminfo`.  Another option would be to find a way to get `musl` compiler to build with `terminfo` using `hadrian`.

Reply:
Oh. Do you have the build failure log?

Reply:
This is fixed now (stage1 terminfo is built on musl).